From bf2bf11d4f2eca66c91dc90b7337efd1eb286efc Mon Sep 17 00:00:00 2001
From: "po.cheng" <po.cheng@adlinktech.com>
Date: Thu, 23 Jan 2025 16:32:39 +0800
Subject: [PATCH] Add Falcon Mode support for iMX8M [PATCH] From
 bf02dd608002ea6a32255db6eca9e9af96764557 Mon Sep 17 00:00:00 2001 From: Elena
 Popa <elena.popa@nxp.com> Date: Wed, 11 Sep 2024 18:20:19 +0300 Subject:
 [PATCH] Add Falcon Mode support for iMX8M and iMX9

Signed-off-by: po.cheng <po.cheng@adlinktech.com>
---
 plat/imx/imx8m/imx8mm/imx8mm_bl31_setup.c | 5 +++++
 plat/imx/imx8m/imx8mn/imx8mn_bl31_setup.c | 5 +++++
 plat/imx/imx8m/imx8mp/imx8mp_bl31_setup.c | 5 +++++
 3 files changed, 15 insertions(+)

diff --git a/plat/imx/imx8m/imx8mm/imx8mm_bl31_setup.c b/plat/imx/imx8m/imx8mm/imx8mm_bl31_setup.c
index 8702d5160..a900e84f8 100644
--- a/plat/imx/imx8m/imx8mm/imx8mm_bl31_setup.c
+++ b/plat/imx/imx8m/imx8mm/imx8mm_bl31_setup.c
@@ -208,6 +208,11 @@ void bl31_early_platform_setup2(u_register_t arg0, u_register_t arg1,
 	 */
 	bl33_image_ep_info.pc = PLAT_NS_IMAGE_OFFSET;
 	bl33_image_ep_info.spsr = get_spsr_for_bl33_entry();
+	// RAM FDT address
+	bl33_image_ep_info.args.arg0 = (u_register_t)0x43000000;
+	bl33_image_ep_info.args.arg1 = 0U;
+	bl33_image_ep_info.args.arg2 = 0U;
+	bl33_image_ep_info.args.arg3 = 0U;
 	SET_SECURITY_STATE(bl33_image_ep_info.h.attr, NON_SECURE);
 
 #if defined(SPD_opteed) || defined(SPD_trusty)
diff --git a/plat/imx/imx8m/imx8mn/imx8mn_bl31_setup.c b/plat/imx/imx8m/imx8mn/imx8mn_bl31_setup.c
index c87748a18..02994e38f 100644
--- a/plat/imx/imx8m/imx8mn/imx8mn_bl31_setup.c
+++ b/plat/imx/imx8m/imx8mn/imx8mn_bl31_setup.c
@@ -220,6 +220,11 @@ void bl31_early_platform_setup2(u_register_t arg0, u_register_t arg1,
 	 */
 	bl33_image_ep_info.pc = PLAT_NS_IMAGE_OFFSET;
 	bl33_image_ep_info.spsr = get_spsr_for_bl33_entry();
+	// RAM FDT address
+	bl33_image_ep_info.args.arg0 = (u_register_t)0x43000000;
+	bl33_image_ep_info.args.arg1 = 0U;
+	bl33_image_ep_info.args.arg2 = 0U;
+	bl33_image_ep_info.args.arg3 = 0U;
 	SET_SECURITY_STATE(bl33_image_ep_info.h.attr, NON_SECURE);
 
 #if defined(SPD_opteed) || defined(SPD_trusty)
diff --git a/plat/imx/imx8m/imx8mp/imx8mp_bl31_setup.c b/plat/imx/imx8m/imx8mp/imx8mp_bl31_setup.c
index 348f90045..99a625eb7 100644
--- a/plat/imx/imx8m/imx8mp/imx8mp_bl31_setup.c
+++ b/plat/imx/imx8m/imx8mp/imx8mp_bl31_setup.c
@@ -241,6 +241,11 @@ void bl31_early_platform_setup2(u_register_t arg0, u_register_t arg1,
 	 */
 	bl33_image_ep_info.pc = PLAT_NS_IMAGE_OFFSET;
 	bl33_image_ep_info.spsr = get_spsr_for_bl33_entry();
+	// RAM FDT address
+	bl33_image_ep_info.args.arg0 = (u_register_t)0x43000000;
+	bl33_image_ep_info.args.arg1 = 0U;
+	bl33_image_ep_info.args.arg2 = 0U;
+	bl33_image_ep_info.args.arg3 = 0U;
 	SET_SECURITY_STATE(bl33_image_ep_info.h.attr, NON_SECURE);
 
 #if defined(SPD_opteed) || defined(SPD_trusty)
-- 
2.25.1

